File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/repository/OauthAuthorizationConsentRepository.java
package io.github.lishangbu.avalon.authorization.repository;

import io.github.lishangbu.avalon.authorization.entity.OauthAuthorizationConsent;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

/// 用户授权确认表(oauth_authorization_consent)数据库访问层
///
/// 提供基础的 CRUD 操作
///
/// @author lishangbu
/// @since 2025/9/14
@Repository
public interface OauthAuthorizationConsentRepository
        extends JpaRepository<
                        OauthAuthorizationConsent,
                        OauthAuthorizationConsent.AuthorizationConsentId>,
                JpaSpecificationExecutor<OauthAuthorizationConsent> {
    Optional<OauthAuthorizationConsent> findByRegisteredClientIdAndPrincipalName(
            String registeredClientId, String principalName);

    void deleteByRegisteredClientIdAndPrincipalName(
            String registeredClientId, String principalName);
}

File: avalon-modules/avalon-authorization/src/main/kotlin/io/github/lishangbu/avalon/authorization/repository/OauthAuthorizationConsentRepository.kt
package io.github.lishangbu.avalon.authorization.repository

import io.github.lishangbu.avalon.authorization.entity.OauthAuthorizationConsent
import io.github.lishangbu.avalon.authorization.entity.OauthAuthorizationConsentId
import org.babyfish.jimmer.spring.repository.KRepository

/**
 * OAuth 授权同意仓储接口
 *
 * 定义OAuth 授权同意数据的查询与持久化操作
 *
 * @author lishangbu
 * @since 2025/9/14
 */
interface OauthAuthorizationConsentRepository : KRepository<OauthAuthorizationConsent, OauthAuthorizationConsentId> {
    /** 按注册客户端 ID 和主体名称查询 OAuth 授权同意 */
    fun findByRegisteredClientIdAndPrincipalName(
        registeredClientId: String,
        principalName: String,
    ): OauthAuthorizationConsent? =
        findNullable(
            OauthAuthorizationConsentId {
                this.registeredClientId = registeredClientId
                this.principalName = principalName
            },
        )

    /** 按注册客户端 ID 和主体名称删除 OAuth 授权同意 */
    fun deleteByRegisteredClientIdAndPrincipalName(
        registeredClientId: String,
        principalName: String,
    ) {
        deleteById(
            OauthAuthorizationConsentId {
                this.registeredClientId = registeredClientId
                this.principalName = principalName
            },
        )
    }
}
